### PR TITLE
Loki: improve proxy detection

### PR DESCRIPTION
**Pull request by mandre**
Not all jobs that use a proxy have `proxy` in their names. Some of the OpenStack jobs (for example External LB) use provider networks and need a proxy to access them.

This commit changes the conditional to detect proxied jobs by relying on the presence of the `proxy-conf.sh` file rather than the name of the job.

### CODE REVIEW

**Comment by mandre** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-externallb

**Comment by mandre** [comment]
This causes the [e2e-openstack-externallb job](https://prow.ci.openshift.org/job-history/gs/origin-ci-test/logs/periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-externallb) to permafail.

Let's see if this patch fixes it.

**Comment by dulek** [comment]
/lgtm

Looks good, that's how we do it in other tests.

**Comment by dgoodwin** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-proxy periodic-ci-openshift-release-master-ci-4.14-e2e-aws-ovn-proxy

**Comment by EmilienM** [comment]
I'll have a look at it tomorrow morning when back on my computer. I need to double check something before it merges.

**Comment by EmilienM** [comment]
aws-ovn-proxy failed for another reason not related to this PR, we can see that this patch has no effect:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/41567/rehearse-41567-periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-proxy/1683887252985352192/artifacts/e2e-aws-ovn-proxy/ipi-install-hosted-loki/build-log.txt

@dgoodwin this is good to go I think.

**Comment by dgoodwin** [comment]
If you are certain this won't break any proxy jobs, then just "hold cancel".

/approve
/hold

File: ci-operator/step-registry/ipi/install/hosted-loki/ipi-install-hosted-loki-commands.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@ then
   exit 0
 fi
 
-if [[ "$JOB_NAME" =~ .*proxy.* ]]
+if test -f "${SHARED_DIR}/proxy-conf.sh"
```
[the remark this answers is not in the document]
Comment (EmilienM) [reply ci-operator/step-registry/ipi/install/hosted-loki/ipi-install-hosted-loki-commands.sh:19]: I'm fine with a OR, let me add it and add a comment too to explain.

[the remark this answers is not in the document]
Comment (EmilienM) [reply ci-operator/step-registry/ipi/install/hosted-loki/ipi-install-hosted-loki-commands.sh]: done

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.14-e2e-aws-ovn-proxy periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-externallb

**Comment by EmilienM** [comment]
/lgtm

I'll hold cancel once the rehearse looks good.

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.14-e2e-aws-ovn-proxy periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-externallb

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.14-e2e-aws-ovn-proxy periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-externallb

**Comment by EmilienM** [comment]
/hold
I found an unrelated CI issue to the external LB job, that I'll fix before we land this one, so we can ensure that this PR actually works fine.

**Comment by dgoodwin** [comment]
Ty!

**Comment by EmilienM** [comment]
I'm waiting on https://github.com/openshift/release/pull/41742 to merge before I can retest this one.

**Comment by EmilienM** [comment]
periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.13-e2e-openstack-externallb

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.13-e2e-openstack-externallb

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.13-e2e-openstack-externallb-techpreview

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.13-e2e-openstack-externallb-techpreview

**Comment by EmilienM** [comment]
The error on 4.13 is confusing to me, see the logs:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/41567/rehearse-41567-periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.13-e2e-openstack-externallb-techpreview/1684676444061437952/artifacts/e2e-openstack-externallb-techpreview/load-balancer/build-log.txt

Not sure what's wrong, but I believe we should backport the installer image change for openstack/upi, so the 4.13 should be green as the 4.14 currently is.

https://github.com/openshift/installer/pull/7374

**Comment by mandre** [comment]
/hold
We need to remove [the last commit](https://github.com/openshift/release/pull/41567/commits/8f6fcfc75c32dcde0ccea6ecb72a17cf522dcbf6) from the PR before this merges.

**Comment by EmilienM** [comment]
now waiting for Dockerfile ansible-core 2.10 backports in installer/4.13 to merge

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-proxy

**Comment by EmilienM** [comment]
/lgtm
/hold cancel

This is good to go.

**Pull request merged**